cicd/forgeops-tests/tests/conftest.py

```diff
@@ -1,13 +1,100 @@
 # Lib imports
 import os
+from requests import get, post
+from utils import logger, rest
+from ProductConfig import IDMConfig, AMConfig, IGConfig, DSConfig
+
+
+root_dir = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..'))
 
 
 def pytest_configure(config):
-    if 'TESTS_NAMESPACE' in os.environ:
-        config._metadata['TESTS_NAMESPACE'] = os.environ['TESTS_NAMESPACE']
+    properties = get_tests_properties()
+    if 'TESTS_NAMESPACE' in properties:
+        config._metadata['TESTS_NAMESPACE'] = properties['TESTS_NAMESPACE']
     else:
         config._metadata['TESTS_NAMESPACE'] = 'smoke'
-    if 'TESTS_DOMAIN' in os.environ:
-        config._metadata['TESTS_DOMAIN'] = os.environ['TESTS_DOMAIN']
+    if 'TESTS_DOMAIN' in properties:
+        config._metadata['TESTS_DOMAIN'] = properties['TESTS_DOMAIN']
     else:
-        config._metadata['TESTS_DOMAIN'] =  'forgeops.com'
+        config._metadata['TESTS_DOMAIN'] = 'forgeops.com'
+
+    if 'TESTS_COMPONENTS' in properties:
+        config._metadata['TESTS_COMPONENTS'] = properties['TESTS_COMPONENTS']
+        version_info = get_version_info(properties['TESTS_COMPONENTS'])
+        for key, value in version_info.items():
+            config._metadata[key] = value
+            properties[key] = value
+
+    report_path = 'reports'
+    if not os.path.exists(report_path):
+        os.makedirs(report_path)
+
+    set_allure_environment_props(os.path.join(root_dir, report_path, 'allure-files', 'environment.properties'),
+                                 properties)
```

Comment:
Does the filename need to be parameterised i.e. we could just pass in 'properties'?